classes/Hook.php

```diff
@@ -235,11 +236,111 @@ public static function getHookAliasList()
     }
 
     /**
+     * Get the list of hook aliases
+     *
+     * @since 1.7.1.0
+     * @return array
+     */
+    private static function getHookAliasesList()
+    {
+        $cacheId = 'hook_aliases';
+        if (!Cache::isStored($cacheId)) {
+            $hookAliasList = Db::getInstance()->executeS('SELECT * FROM `'._DB_PREFIX_.'hook_alias`');
+            $hookAliases = array();
+            if ($hookAliasList) {
+                foreach ($hookAliasList as $ha) {
+                    if (!isset($hookAliases[$ha['name']])) {
+                        $hookAliases[$ha['name']] = array();
+                    }
+                    $hookAliases[$ha['name']][] = $ha['alias'];
+                }
+            }
+            Cache::store($cacheId, $hookAliases);
+            return $hookAliases;
+        }
+        return Cache::retrieve($cacheId);
+    }
+
+    /**
+     * Return backward compatibility hook names
+     *
+     * @since 1.7.1.0
+     * @param $hookName
+     * @return array
+     */
+    private static function getHookAliasesFor($hookName)
+    {
+        $cacheId = 'hook_aliases_' . $hookName;
+        if (!Cache::isStored($cacheId)) {
+            $aliasesList = Hook::getHookAliasesList();
+
+            if (isset($aliasesList[$hookName])) {
+                Cache::store($cacheId, $aliasesList[$hookName]);
+                return $aliasesList[$hookName];
+            }
+
+            $retroName = array_keys(array_filter($aliasesList, function ($elem) use ($hookName) {
+               return in_array($hookName, $elem) ;
+            }));
+
+            if (empty($retroName)) {
+                Cache::store($cacheId, array());
+                return array();
+            }
+
+            Cache::store($cacheId, $retroName);
+            return $retroName;
+        }
+        return Cache::retrieve($cacheId);
+    }
+
+    /**
+     * Check if a hook or one of its old names is callable on a module
+     *
+     * @since 1.7.1.0
+     * @param $module
+     * @param $hookName
+     * @return bool
+     */
+    private static function isHookCallableOn($module, $hookName)
+    {
+        $aliases = Hook::getHookAliasesFor($hookName);
+        $aliases[] = $hookName;
+
+        return array_reduce($aliases, function ($prev, $curr) use ($module) {
+            return $prev || is_callable(array($module, 'hook' . $curr));
```

Comment:
So, as soon as a hook is callable, all the other aliases will be considered as callable as well ?

To be checked because I wonder if we may bring an error here, something like a call to an undefined function.